app/controllers/pets_controller.rb

```diff
@@ -36,11 +36,21 @@ def destroy
   end
 
   def change_application_status
-    pet = Pet.find(params[:id])
     approved_applicant_name = Application.find(params[:approved_applicant]).name
-    pet.update(approved_applicant: approved_applicant_name)
-    pet.change_to_pending
-    redirect_to "/pets/#{pet.id}"
+    application_id = Application.find(params[:approved_applicant]).id
```

Comment:
Added the ability to find application by id as well as name based on single or multiple selection.